src/jit/simdcodegenxarch.cpp

```diff
@@ -493,7 +493,9 @@ void CodeGen::genSIMDScalarMove(var_types type, regNumber targetReg, regNumber s
                         // In general, when we use a three-operands move instruction, we want to merge the src with
                         // itself. This is an exception in that we actually want the "merge" behavior, so we must
                         // specify it with all 3 operands.
-                        inst_RV_RV_RV(ins, targetReg, targetReg, srcReg, emitTypeSize(targetType));
```

Comment:
>emitTypeSize(targetType)); [](start = 73, length = 26)

We are supposed to use emitTypeSize(type) and type here.

Same comment applies to lines 502, 524, 542 and 566 as well.

In general since this routine is meant for scalar move, it should be using type/emitTypeSize(type)  instead of targetType/emitTypeSize(targetType) and should hard code specific SIMD type while emitting insertps and shift left.
We can completely do away with targetType.